src/games/strategy/triplea/ai/proAI/util/ProBattleUtils.java

```diff
@@ -207,17 +215,17 @@ public ProBattleResult callBattleCalculator(final PlayerID player, final Territo
     final int runCount = Math.max(16, 100 - minArmySize);
     if (isAttacker) {
       if (retreatWhenOnlyAirLeft) {
-        ai.getCalc().setRetreatWhenOnlyAirLeft(true);
+        calc.setRetreatWhenOnlyAirLeft(true);
       }
       results =
-          ai.getCalc().setCalculateDataAndCalculate(player, t.getOwner(), t, attackingUnits, defendingUnits,
+          calc.setCalculateDataAndCalculate(player, t.getOwner(), t, attackingUnits, defendingUnits,
               new ArrayList<Unit>(bombardingUnits), TerritoryEffectHelper.getEffects(t), runCount);
       if (retreatWhenOnlyAirLeft) {
-        ai.getCalc().setRetreatWhenOnlyAirLeft(false);
+        calc.setRetreatWhenOnlyAirLeft(false);
```

Comment:
The above refactor would help clarify this method. If it's called "doNotRetreat", or "retreatWhenLosing", it will make the functionality of this method crystal clear. As is, you have to assume it does the exact same as the check box on the battle calculator, which is a leap of faith.